contrib/ledger-tests/run_and_alert.sh

```diff
@@ -0,0 +1,168 @@
+# Example: `gs://...`
+
+# 11. GCLOUD_KEY_FILE
+# Description: Path to the Google Cloud key file used for authentication.
+
+# -----------------------------------------------------------------------------
+# Cleanup previous run
+
+rm -rf $LEDGER_DIR && mkdir $LEDGER_DIR
+rm -rf $LEDGER_MIN_DIR && mkdir $LEDGER_MIN_DIR
+
+# -----------------------------------------------------------------------------
+# Setup
+START_TIME=$(date +%s)
+
+cd $FIREDANCER_DIR
+rm -rf $NETWORK-*.tar.gz
+git checkout $REPO_BRANCH
+git pull origin $REPO_BRANCH
+PATCH_FILES=""
+
+for patch in "$PATCH_DIR"/*.diff;
+do
+  if [ -f "$patch" ]; then
+    git apply "$patch"
+    PATCH_FILES+="$(basename "$patch") "
+  fi
+done
+
+GIT_COMMIT=$(git rev-parse HEAD)
+
+PATH=/opt/rh/gcc-toolset-13/root/usr/bin:$PATH
+export PATH
+PKG_CONFIG_PATH=/usr/lib64/pkgconfig:$PKG_CONFIG_PATH
+
+make distclean && make clean
+./deps.sh nuke
+echo "y" | FD_AUTO_INSTALL_PACKAGES=1 ./deps.sh +dev
+make -j
+
+prlimit --pid=$$ --nofile=1048576
+prlimit --pid=$$ --memlock=unlimited
+
+gcloud auth activate-service-account --key-file=$GCLOUD_KEY_FILE
+
+# -----------------------------------------------------------------------------
+# Run
+
+slack_alert() {
+    local alert_message=$1
+    local ledger_min_basename=$(basename "$LEDGER_MIN_DIR")
+    local metadata=$FIREDANCER_DIR/dump/$ledger_min_basename/metadata
+
+    local mismatch_slot=$(echo "$alert_message" | grep -oP 'mismatch_slot: \K\d+')
+    local replay_start_slot=$(grep 'replay_start_slot=' $metadata | cut -d'=' -f2)
+    local replay_time=$(grep 'replay_time=' $metadata | cut -d'=' -f2)
+    local replay_slots_before_mismatch=$((mismatch_slot - replay_start_slot))
+    
+    curl -X POST -H 'Content-type: application/json' --data "{\"text\":\"Network: $NETWORK \nCommit: $GIT_COMMIT \nPatches: $PATCH_FILES \nAlert: $alert_message \nURL: $UPLOAD_URL/$NETWORK-$mismatch_slot.tar.gz \nSlots Replayed: $replay_slots_before_mismatch \nReplay Time: $replay_time's \"}" $SLACK_WEBHOOK_URL
+}
+
+alert_success() {
+    local end_info=$1
+    local end_slot=$(echo "$end_info" | grep -oE '[0-9]{9}' | tail -n 1)
+    local ledger_min_basename=$(basename "$LEDGER_MIN_DIR")
+    local metadata=$FIREDANCER_DIR/dump/$ledger_min_basename/metadata
+
+    local alert_message="Ledger Test Success"
+    local replay_start_slot=$(grep 'replay_start_slot=' $metadata | cut -d'=' -f2)
+    local replay_time=$(grep 'replay_time=' $metadata | cut -d'=' -f2)
+    local replay_slots_before_success=$((end_slot - replay_start_slot))
+
+    curl -X POST -H 'Content-type: application/json' --data "{\"text\":\"Network: $NETWORK \nCommit: $GIT_COMMIT \nPatches: $PATCH_FILES \nAlert: $alert_message \nSlots Replayed: $replay_slots_before_success \nReplay Time: $replay_time's \"}" $SLACK_WEBHOOK_URL
+}
+
+cd $FIREDANCER_DIR/contrib/ledger-tests
+
+if [[ "$NETWORK" == "mainnet" ]]; then
+    $FIREDANCER_DIR/contrib/ledger-tests/ledger_conformance.sh all \
```

Comment:
do we want to run `ledger_conformance`? seems like protobuf dumping isnt 100% correct right now as of a couple weeks ago (0 protobufs failed for that mismatch we looked at)